src/index.js

```diff
@@ -88,9 +88,18 @@ const ChartComponent = React.createClass({
 		nextData.forEach(function (dataset, sid) {
 			if (currentData[sid] && currentData[sid].data) {
 				currentData[sid].data.splice(nextData[sid].data.length);
+
 				dataset.data.forEach(function (point, pid) {
 					currentData[sid].data[pid] = nextData[sid].data[pid];
 				});
+
+				delete dataset.data;
```

Comment:
hmm, I'm not sure we are building the new version of data set on each render.

Regardless, if we can just do:
`const { data, ...dataset } = dataset` instead of `delete dataset.data;`

I would be really happy 😃